ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/LocalMediaPlayer.kt

```diff
@@ -125,6 +125,10 @@ class LocalMediaPlayer(
     }
 
     fun release() {
+        // Calling reset() will result in changing this player's state. If we allow
+        // the onPlayerStateChanged callback, then the state change will cause this
+        // to resurrect the media session which has just been destroyed.
+        onPlayerStateChanged = null
```

Comment:
See the comment, I think this here is a fix for a bug that exists regardless of Android Auto. 

When the Service is being shutdown it calls `MediaPlayerService::onDestroy()` which calls this function here. When this function `reset()` below, it sets the play's state to IDLE. But the act of setting the state will call a callback, which will then start [re-creating the Media Session](https://github.com/ultrasonic/ultrasonic/blob/develop/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaPlayerService.kt#L371). Then the `MediaPlayerService::onDestroy()` will resume and continue destroying the Media Session.

This is all bad behavior for the existing app, but for Android Auto we have to set the session token, which can only be done once. All this destroying and re-creating ends up attempting to set it twice, which results in an exception. By setting `onPlayerStateChanged` to `null` here we remove the callback and allow this thing to die gracefully.